src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
+        struct vt_stat vcs = {};
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+
+        if (fd >= 0) {
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
+                        return false;
+                }
+                idx = vcs.v_active;
+        }
+
+        xsprintf(vcname, "/dev/vcs%i", idx);
+        if (access(vcname, F_OK) < 0)
+                return false;
+        return true;
+}
```

Comment:
For the record - if we stick to returning errors instead of logical yes/no from is_*() functions, what should I return in is_settable() - if ioctl() succeeds (basically always) but console is not in xlate/unicode mode ? As I have to fake the error in such case.